login.js

```diff
@@ -1,28 +1,67 @@
 import storage from './storage.js';
-import { storageHost } from './constants.js'
+import {getContractSource, hexToWordList, wordListToHex} from './blockchain.js';
+import {storageHost} from './constants.js'
 
 const loginEndpoint = 'https://login.exokit.org';
-const usersEndpoint = 'https://users.exokit.org';
+// const usersEndpoint = 'https://users.exokit.org';
 
 const _clone = o => JSON.parse(JSON.stringify(o));
 
 let loginToken = null;
 let userObject = null;
-async function pullUserObject() {
-  const res = await fetch(`${usersEndpoint}/${loginToken.name}`);
-  if (res.ok) {
-    userObject = await res.json();
-  } else if (res.status === 404) {
-    userObject = {
-      name: loginToken.name,
-      avatarHash: null,
-      inventory: [],
-    };
-  } else {
-    throw new Error(`invalid status code: ${res.status}`);
+async function ensureUserObjectBaked() {
+  const contractSource = await getContractSource('isUserAccountBaked.cdc');
+
+  const res = await fetch(`https://accounts.exokit.org/sendTransaction`, {
+    method: 'POST',
+    body: JSON.stringify({
+      limit: 100,
+      script: contractSource.replace(/ARG0/g, '0x' + loginToken.addr),
+      wait: true,
+    }),
+  });
+  const response = await res.json();
+  const isBaked = response.encodedData.value;
+  if (!isBaked) {
+    const contractSources = await getContractSource('bakeUserAccount.json');
+    for (const contractSource of contractSources) {
```

Comment:
contractSources can be null

Reply:
That would be a serious error in the contracts deployment that we probably can't recover from anyway, so I'm fine with that crashing here.